docs/src/pages/customization/theme-components/theme-components.md

```diff
@@ -29,31 +52,45 @@ The list of each component's classes is documented under the **CSS** section of
 
 To override a lab component's styles with TypeScript, check [this section of the documentation](/components/about-the-lab/#typescript).
 
-## Default props
+### Overrides based on props
 
-You can change the default of every prop of a MUI component.
-A `defaultProps` key is exposed in the theme's `components` key for this use case.
+You can pass a callback as a value in each slot of the component's `styleOverrides` to apply styles based on props.
+
+The `ownerState` prop is a combination of public props that you pass to the component + internal state of the component.
 
 ```js
-const theme = createTheme({
+const finalTheme = createTheme({
   components: {
-    // Name of the component
-    MuiButtonBase: {
-      defaultProps: {
-        // The props to change the default for.
-        disableRipple: true, // No more ripple!
+    MuiSlider: {
+      styleOverrides: {
+        valueLabel: ({ ownerState, theme }) => ({
+          ...(ownerState.orientation === 'vertical' && {
+            backgroundColor: 'transparent',
+            color: theme.palette.grey[500],
+          }),
+        }),
       },
     },
   },
 });
 ```
 
-{{"demo": "pages/customization/theme-components/DefaultProps.js"}}
+{{"demo": "pages/customization/theme-components/GlobalThemeOverrideCallback.js"}}
 
-To override lab component styles with TypeScript, check [this page](/components/about-the-lab/#typescript).
+### Using `sx` (experimental) syntax
+
+If you are not familiar `sx`, first check out [the concept](/system/the-sx-prop) and [the difference with the `styled`](/system/styled/#difference-with-the-sx-prop).
+
+`sx` is also compatible with theme style overrides if you prefer the shorthand notation.
+
+{{"demo": "pages/customization/theme-components/GlobalThemeOverrideSx.js"}}
 
 ## Adding new component variants
 
+> ⚠️ This API has been **deprecated** and will likely be removed in the next major release. If you want to apply styles based on props, take a look at [Overrides based on props](#overrides-based-on-props) instead.
```

Comment:
I landed here after seeing the notion of deprecation in #30668. 

We might be jumping one step head too quickly on this one. I'm adding a new comment on why we added the variant API in the past, it doesn't seem that the true purpose in #30412 or in #28107 (from my perspective on why we added it in the first place, it wasn't because of JSS, when we added it we were already adding emotion from what I recall, we knew this PR was going to be possible of it is covered.

Reply:
Okay, I will remove the deprecation section for now and update the #30412.